--- a/ethersrv-linux.c
+++ b/ethersrv-linux.c
@@ -34,13 +34,14 @@
 #include <linux/if_ether.h>
 #include <linux/if_packet.h>
 #include <limits.h>          /* PATH_MAX and such */
-#include <net/ethernet.h>
 #include <net/if.h>
 #include <signal.h>
 #include <stdio.h>
 #include <string.h>          /* mempcy() */
 #include <sys/ioctl.h>
+#include <sys/select.h>
 #include <sys/socket.h>
+#include <sys/time.h>
 #include <stdint.h>          /* uint16_t, uint32_t */
 #include <stdlib.h>          /* realpath() */
 #include <time.h>            /* time() */
